src/protocol/utils.ts

```diff
@@ -131,8 +131,8 @@ export const EventEmitter = {
       );
     };
 
-    obj.emit = (name, value) => {
-      (obj.eventListeners.get(name) || []).forEach(handler => handler(value));
+    obj.emit = (name, ...values) => {
+      (obj.eventListeners.get(name) || []).forEach(handler => handler(...values));
```

Comment:
Is this going to be okay? Won't we have to change all handler function paramaters to expect an array because of the spread operator change?